test/cql-pytest/test_guardrail_replication_strategy.py

```diff
@@ -146,3 +146,12 @@ def test_given_restrict_replication_simplestrategy_when_it_is_set_should_emulate
         config_modifications.enter_context(config_value_context(cql, 'restrict_replication_simplestrategy', 'warn'))
         create_ks_and_assert_warnings_and_errors(cql, f" WITH REPLICATION = {{ 'class' : 'SimpleStrategy', 'replication_factor' : 3 }}",
                                                  warnings_count=1, fails_count=0)
+
+def test_config_replication_strategy_warn_list_roundtrips_quotes(cql):
+    # Use direct SELECT/UPDATE to avoid trippy config_value_context behavior
+    value = cql.execute("SELECT value FROM system.config WHERE name = 'replication_strategy_warn_list'").one().value
+    assert value == '["SimpleStrategy"]' # our lovely default
```

Comment:
Why do you need this SELECT or assert at all? This test will neeedless break if in the future we'll change this default (e.g., say we want to also warn against XyzStrategy).
Basically, you can remove those two lines from this test, and have a test that confirms that the names of the strategies can be given either quoted or unquoted.

Reply:
This ensures that the test as a whole doesn't dirty the cluster.

Reply:
Good point, you can keep the SELECT to read the config value, as you did, and then try to write back the *same* value and see that it works. This would _really_ be a "round trip" test. 

But I actually prefer what you did that isn't about "round trip" at all, but in fact about testing that both with and without quotes always works (regardless of whether the default uses quotes or not). You can achieve that without relying on the default by (assuming, as we do, in all tests involving configuration, that the cluster isn't used in parallel) saving the config parameter before writing garbage into it - and then restore it. We even have a utility function config_value_context to do this easily (change the config and restore it at the end of the scope).

I'm just afraid it will be annoying in the future to fix this test if we ever change the default of this option, because of the invention of some other replication strategy that has absolutely nothing with this test.

Reply:
That's okay, I don't expect huge traffic on that configuration option.

Reply:
@nyh any more adjustments needed?

Reply:
@nyh help

Reply:
@nyh how can we move this forward?

Reply:
As I said, I'm not happy for a test of one feature (basically - the ability to allow double quotes on option string) to depend on a different feature - here the default of some guardrail (after all, the whole raison d'etre of having a configurable guardrail is the ability to change it!). But it's not important enough to hold this patch, and if we'll have problems with this test we can easily fix it later. So I'll merge.